Realm/RLMQueryUtil.mm

```diff
@@ -207,10 +207,18 @@ void add_string_constraint_to_query(realm::Query &query,
     }
 }
 
+void add_string_constraint_to_query(realm::Query &query,
+                                    NSPredicateOperatorType operatorType,
+                                    NSComparisonPredicateOptions predicateOptions,
+                                    Columns<String> &&column,
+                                    id value) {
+    add_string_constraint_to_query(query, operatorType, predicateOptions, std::forward<Columns<String>>(column), RLMStringDataWithNSString(value));
+}
+
 void add_string_constraint_to_query(realm::Query& query,
                                     NSPredicateOperatorType operatorType,
                                     NSComparisonPredicateOptions predicateOptions,
-                                    NSString *value,
+                                    id value,
```

Comment:
I find the fact that this change is needed somewhat unsettling, but looking at what breaks with it being `NSString *` I don't see an obvious solution.